addon/ng2/utilities/module-resolver.ts

```diff
@@ -0,0 +1,113 @@
+import * as ts from 'typescript';
+import * as dependentFilesUtils from './get-dependent-files';
+
+import { Promise } from 'es6-promise';
+import { Change, ReplaceChange } from './change';
+
+/** @const The root directory of Angular Project. */
+const ROOT_PATH = path.resolve('src/app');
+
+/** 
+ * @class Rewrites import module of dependent files when the file is moved. 
+ *        Also rewrites export module of related index file of the given file.
+ */
+export class ModuleResolver {
+
+  /** @member Description of the module resolution. */
+  description: string;
+
+  /** @constructs */
+  constructor(public oldFilePath: string, public newFilePath: string) {
+    let component = path.basename(oldFilePath);
+    this.description = `Promoted ${component} from ${oldFilePath} to ${newFilePath}`;
+  }
+
+  /**
+   * Changes are applied from the bottom of a file to the top.
+   * An array of Change instances are sorted based upon the order, 
+   * then apply() method is called sequentially.
+   * 
+   * @param changes {Change []} 
+   * @return Promise after all apply() method of Change class is called 
+   *         to all Change instances sequentially.
+   */
+  applySortedChangePromise(changes: Change[]): Promise<void> {
+    return changes
+      .sort((currentChange, nextChange) => nextChange.order - currentChange.order)
+      .reduce((newChange, change) => newChange.then(() => change.apply()), Promise.resolve());
+  }
+
+  /** 
+   * Rewrites the import specifiers of all the dependent files (cases for no index file)
+   * 
+   * @todo Implement the logic for rewriting imports of the dependent files when the file
+   *       being moved has index file in its old path and/or in its new path.
+   * 
+   * @return {Promise<Change[]>} 
+   */
+  resolveDependentFiles(): Promise<Change[]> {
+    return dependentFilesUtils.getDependentFiles(this.oldFilePath, ROOT_PATH)
+      .then((files:  dependentFilesUtils.ModuleMap) => {
+        let changes: Change[] = [];
+        Object.keys(files).forEach(file => {
+          let tempChanges: ReplaceChange[] = files[file]
+            .map(specifier => {
+              let fileDir = path.dirname(file);
+              let changeText = path.relative(fileDir, this.newFilePath);
+              if (changeText.length > 0 && changeText.charAt(0) !== '.') {
+                changeText = `.${path.sep}${changeText}`;
+              }
+              if (changeText === '') {
+                changeText = `.${path.sep}${path.basename(this.oldFilePath, '.ts')}`;
+              } else {
+                changeText = `${changeText}${path.sep}${path.basename(this.oldFilePath, '.ts')}`;
+              }
+              let position = specifier.end - specifier.specifierText.length;
+              return new ReplaceChange(file, position - 1, specifier.specifierText, changeText);
+            });
+          if (tempChanges.length > 1) {
+            tempChanges = tempChanges.sort((currentChange, nextChange) => {
```

Comment:
Extract the comparator function you have into a named function - that will make this read a lot better:

`tempChanges.sort(byChangeOrder)`